### DISCUSSION TITLE
Custom field grab IP of interface

### DISCUSSION THREAD

**Discussion by jfreak53** [netbox-community/netbox#12971] [Help Wanted!]
How do I grab the IP address of one of my interfaces on a device? I have one interface on certain servers call 'idrac-1', I only ever assign 1 IP to it, but how do I grab that in a custom link? I've tried the below code, it doesn't grab anything:

`{{ object.interfaces.idrac-1.ip }}`

**Comment by AADIP** [comment]
Typically I would use `object.primary_ip4.address.ip`  if the IP address were the primary for the device object. 

Would something like this help you:
```
{% set interface = object.interfaces.get(name='idrac-1') %}
{% set ip_address = interface.ip_addresses.first()|string() %}
https://{{ ip_address[:-3] }}
```
I had to set a string-casted variable and slice separately but it seems to work for me in testing.

**Reply by AADIP** [reply]
I realized that the string slicing might not be the best once you go under a /10 subnet. Also, I haven't tried this but you may be able to access the IP address itself from `ip_addresses`.

**Reply by candlerb** [reply]
If you have marked your idrac interface as the only "mgmt" interface (or it has a fixed name), then: https://github.com/netbox-community/netbox/discussions/11752#discussioncomment-4995855
